homeassistant/components/microbees/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for microBees integration."""
+import logging
+from typing import Any
+
+from microBeesPy.exceptions import MicroBeesWrongCredentialsException
+from requests.exceptions import ConnectTimeout, HTTPError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_EMAIL,
+    CONF_PASSWORD,
+    CONF_TOKEN,
+)
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+from .microbees import MicroBeesConnector
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_CLIENT_ID): str,
+        vol.Required(CONF_CLIENT_SECRET): str,
```

Comment:
Does this use Oauth?

Reply:
Actually yes, but token is long life so don't need to refresh

Reply:
Wouldn't it be nicer if we could use the Oauth flow instead of storing the credentials of users?

Reply:
Actually don't store credentials of user  but user Oauth cliend_id and secret and after this the token generated. Alternatively we have to use a unique client_id and secret for all the users and this is not good for us because we have limits on monthly call by client_id

Reply:
We have an Oauth flow, checkout integrations like google, withings, spotify etc. People fill in their own client id and secret and will be redirected to the service to login. After that they will be redirected back to HA.

The current approach would also work, but I think this approach would be a better user experience, and it would make HA a bit less dependent on the decisions in the company. (Maybe you will make new tokens short lifed, etc)

Reply:
Yes, we've seen the other integrations approach. OK, we will try to use that, is this a now decision or we can do in a second version of the integration?

Reply:
I asked around for a second opinion, maybe this approach is a good reason, but I want to be 100% sure what I am telling is true.

If we were to implement application_credentials, it should be in this PR

Reply:
It's not a implicit flow, is Password Credentials Grant type

Reply:
Hi @joostlek , we've choosed to leave Password Credentials Grant Type because is legacy. Now we use implicit, we've developed the flow as suggested from HA. The integration looks now ready to be reviewed